source/core/index.ts

```diff
@@ -262,6 +262,10 @@ const waitForOpenFile = async (file: ReadStream): Promise<void> => new Promise((
 		reject(error);
 	};
 
+	if (!file.pending) {
+		resolve();
+	}
+
```

Comment:
Slightly unrelated, but shouldn't we listen to `ready` instead of `open`?

```
Event: 'ready'#

Added in: v9.11.0
Emitted when the fs.ReadStream is ready to be used.

Fires immediately after 'open'.
```

Probably will not change much, but maybe worth it just in case.

Reply:
I don't know why there's that code, but I think @szmarczak wanted to check if there was any error opening the file (eg. file does not exists), so `open` seems to be the best choice.

BTW: reading this "Fires immediately after 'open'." I think `open` and `ready` are completely equivalent, I think it's some guy at Node.JS that just anted to mess with the users.

Reply:
I’m asking as `.pending` docs mention the `ready` event, not `open`.

Reply:
The Node.JS documentation make no sense:
`open`: Emitted when the fs.ReadStream's file descriptor has been opened.
`ready`: Emitted when the fs.ReadStream is ready to be used.
`readStream.pending`: "This property is true if the underlying file has not been opened yet, i.e. before the 'ready' event is emitted."

`pending` should be "This property is true if the underlying file has not been opened yet, i.e. before the 'open' event is emitted."

Looking at the code:
`pending` is defined as `this.fd === null` (https://github.com/nodejs/node/blob/master/lib/internal/fs/streams.js#L265,L268)
The `open` event is fired right after the assignation of `fd` and the `ready` event is fired right after the `open` event.
(https://github.com/nodejs/node/blob/master/lib/internal/fs/streams.js#L151,L153)

Reply:
I would recommend opening a Node.js issue about it. The docs should be clear about which event to use for what. It's currently very unclear.

Reply:
Let's for with `ready` to be consistent.